src/test/java/io/tiledb/spark/TileDBDataSourceReadTestBufferSizes.java

```diff
@@ -77,39 +77,40 @@ public void testQuickStartSparseWith4ByteBufferIncompleteReallocDisabled() {
     }
   }*/
 
-  /**
-   * This test uses an 4 byte buffer which is too small to hold any results, buffer reallocation is
-   * disabled so we should get an exception
-   */
-  @Test
-  public void testQuickStartSparseWith4ByteBufferIncomplete() {
-    Dataset<Row> dfRead =
-        session()
-            .read()
-            .format("io.tiledb.spark")
-            .option("uri", testArrayURIString("quickstart_sparse_array"))
-            .option("read_buffer_size", 4)
-            .option("allow_read_buffer_realloc", true)
-            .load();
-    dfRead.createOrReplaceTempView("tmp");
-    List<Row> rows = session().sql("SELECT * FROM tmp").collectAsList();
-
-    Assert.assertEquals(3, rows.size());
-    // A[1, 1] == 1
-    Row row = rows.get(0);
-    Assert.assertEquals(1, row.getInt(0));
-    Assert.assertEquals(1, row.getInt(1));
-    Assert.assertEquals(1, row.getInt(2));
-    // A[2, 3] == 3
-    row = rows.get(1);
-    Assert.assertEquals(2, row.getInt(0));
-    Assert.assertEquals(3, row.getInt(1));
-    Assert.assertEquals(3, row.getInt(2));
-    // A[2, 4] == 2
-    row = rows.get(2);
-    Assert.assertEquals(2, row.getInt(0));
-    Assert.assertEquals(4, row.getInt(1));
-    Assert.assertEquals(2, row.getInt(2));
-    return;
-  }
+  //  /**
```

Comment:
Yes, there were some issues with buffer reallocation when using ArrowBuffers. I decided to leave it as a known issue since the impact is minor. If the read_buffer_size provided by the user is not big enough a prompt message will ask for an increase. Previously that used to happen automatically by doubling the size of the buffers every time. I will open an issue on Shortcut for this.

Reply:
Thanks, please link and then resolve.